sycl/test/regression/fp16-with-unnamed-lambda.cpp

```diff
@@ -0,0 +1,43 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+#include <CL/sycl.hpp>
+
+#include <iostream>
+
+int main() {
+  auto AsyncHandler = [](cl::sycl::exception_list EL) {
+    for (std::exception_ptr const &P : EL) {
+      try {
+        std::rethrow_exception(P);
+      } catch (std::exception const &E) {
+        std::cerr << "Caught async SYCL exception: " << E.what() << std::endl;
+      }
+    }
+  };
+
+  cl::sycl::queue Q(AsyncHandler);
+
+  cl::sycl::device D = Q.get_device();
+  if (!D.has_extension("cl_khr_fp16"))
+    return 0; // Skip the test if halfs are not supported
+
+  cl::sycl::buffer<cl::sycl::cl_half> Buf(1);
```

Comment:
I assume just `half` type also will work, right?

Reply:
Actually, there is no such thing as `half` according to the SYCL spec, see KhronosGroup/SYCL-CTS#37

But we have such alias in our implementation and it should also work, because `cl::sycl::cl_half` is declared as an alias to `half`

Reply:
Nope. SYCL spec defines `half`. See Table 6.1. https://www.khronos.org/registry/SYCL/specs/sycl-1.2.1.pdf . There is `half` defined.

Reply:
Ok, I see.

However, it says that "all standard C++ fundamental types from Table 6.1", while `half` is not a standard fundamental data type, see [Floating point types](https://en.cppreference.com/w/cpp/language/types)

Reply:
But `half` also is presented in this table.

Reply:
Feel free to fix the spec if you think there are some issues. :-)